### PR TITLE
GitHub actions update @v3 -> @v4

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] github actions

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -139,14 +139,14 @@ jobs:
 
     - name: HTML coverage reports
       if: always()
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
```
Comment (babenek) [inline .github/workflows/test.yml:142]:
```suggestion
      uses: actions/upload-artifact@v3
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -139,14 +139,14 @@ jobs:
...
       with:
         name: coverage_html-${{ matrix.python-version }}
         path: coverage_html
 
     - name: Upload coverage reports to Codecov
       if: ${{ matrix.python-version == '3.8' }}
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@v4
```
Comment (babenek) [inline .github/workflows/test.yml:149]:
```suggestion
      uses: codecov/codecov-action@v3
```

**Review by babenek — COMMENTED** [review]
fix?

**Review by Yullia — APPROVED** [review]
Approved

**Review by iuriimet — APPROVED** [review]

**Pull request merged**